### ISSUE TITLE
[Elements] MRL keyboard navigation not working as expected

### ISSUE THREAD

**Issue by skrustev** [IgniteUI/igniteui-angular#17468]
### Description

When using Multi Row Layout in the Grid, the keyboard navigations is acting not as expected

### Framework

Angular Elements

### Angular Version

22.x

### Ignite UI for Angular Version

22.1.x

### Component / Area

Grid

### Browser

Chrome

### Command Not Working

```shell

```

### Steps to Reproduce

1. Open https://staging.infragistics.com/webcomponents-demos/grids/grid/multi-row-layout-options?nav=0
2. Click on any of the cells and try to navigate up/down/left/right
3. Compare the navigation to https://www.infragistics.com/webcomponents-demos/grids/grid/multi-row-layout-options?nav=0

### Actual Result

Keyboard navigation skips over cells, where it shouldn't and cannot go from pinned to unpinned area and vice versa.

### Expected Result

The keyboard navigation should work as expected.

### Attachments

Expected behavior:
<img width="1523" height="879" alt="Image" src="https://github.com/user-attachments/assets/df43be93-e6e8-44cb-b1f2-e407b61f9232" />

Latest behavior:
<img width="1523" height="879" alt="Image" src="https://github.com/user-attachments/assets/48587da8-488d-4cca-a2ff-2ae54db243ce" />
